litmus-portal/frontend/src/views/Analytics/ApplicationDashboards/Table/index.tsx

```diff
@@ -183,33 +210,81 @@ const DashboardTable: React.FC = () => {
                 ? sortNumAsc(y, x)
                 : sortNumDesc(y, x);
             }
-            if (filter.sortData.dashboardType.sort) {
-              const x = a.db_type;
-              const y = b.db_type;
-              return filter.sortData.dashboardType.ascending
-                ? sortAlphaAsc(x, y)
-                : sortAlphaDesc(x, y);
-            }
-            if (filter.sortData.dataSourceType.sort) {
-              const x = a.ds_type;
-              const y = b.ds_type;
-              return filter.sortData.dataSourceType.ascending
-                ? sortAlphaAsc(x, y)
-                : sortAlphaDesc(x, y);
-            }
-            if (filter.sortData.agent.sort) {
-              const x = a.cluster_name;
-              const y = b.cluster_name;
-              return filter.sortData.agent.ascending
-                ? sortAlphaAsc(x, y)
-                : sortAlphaDesc(x, y);
-            }
             return 0;
           })
     : [];
 
+  useEffect(() => {
+    if (dataSourceList && dataSourceList.ListDataSource) {
+      const activeDataSources: ListDataSourceResponse[] =
+        dataSourceList.ListDataSource.filter(
+          (dataSource) => dataSource.health_status === 'Active'
+        ) ?? [];
+      if (activeDataSources.length) {
+        setActiveDataSourceAvailable(true);
+      }
+    }
+  }, [dataSourceList]);
+
   return (
     <div className={classes.root}>
+      {!activeDataSourceAvailable && !loadingDataSources && (
+        <blockquote className={classes.warningBlock}>
+          <Typography className={classes.warningText} align="left">
+            {dataSourceList?.ListDataSource.length
+              ? t(
+                  'analyticsDashboard.applicationDashboardTable.warning.noActiveDataSource'
+                )
+              : t(
+                  'analyticsDashboard.applicationDashboardTable.warning.noAvailableDataSource'
+                )}
+          </Typography>
+          <div className={classes.warningActions}>
+            {dataSourceList?.ListDataSource.length ? (
+              <>
```

Comment:
avoid empty tags at all places

Reply:
React supports this and a div was not required here to wrap the child components so it has been done please refer to usage in existing code base @whitetiger1399

Reply:
cc: @S-ayanide

Reply:
cc: @arkajyotiMukherjee

Reply:
Fragment is fine here as it has two children...we can make it better by using `&&` instead of ternary @ishangupta-ds

Reply:
tried using that 0 is getting rendered on the screen for negative case @arkajyotiMukherjee